app/src/test/java/org/mozilla/focus/utils/DownloadUtilsTest.java

```diff
@@ -15,42 +15,56 @@
+
     private static void assertContentDisposition(String expected, String contentDisposition) {
         Download download = new Download(null, null, contentDisposition, null, 0, null);
         assertEquals(expected, DownloadUtils.guessFileName(download));
     }
 
     @Test
     public void testGuessFileName_contentDisposition() {
-        // Default file name
+        String contentDisposition = "attachment";
+
+        // Default file name without content disposition
         assertContentDisposition("downloadfile.bin", "");
-        assertContentDisposition("downloadfile.bin", "attachment");
-        assertContentDisposition("downloadfile.bin", "attachment;");
-        assertContentDisposition("downloadfile.bin", "attachment; filename");
-        assertContentDisposition(".bin", "attachment; filename=");
-        assertContentDisposition(".bin", "attachment; filename=\"\"");
-
-        // Provided filename field
-        assertContentDisposition("filename.jpg", "attachment; filename=\"filename.jpg\"");
-        assertContentDisposition("file\"name.jpg", "attachment; filename=\"file\\\"name.jpg\"");
-        assertContentDisposition("file\\name.jpg", "attachment; filename=\"file\\\\name.jpg\"");
-        assertContentDisposition("file\\\"name.jpg", "attachment; filename=\"file\\\\\\\"name.jpg\"");
-        assertContentDisposition("filename.jpg", "attachment; filename=filename.jpg");
-        assertContentDisposition("filename.jpg", "attachment; filename=filename.jpg; foo");
-        assertContentDisposition("filename.jpg", "attachment; filename=\"filename.jpg\"; foo");
-
-        // UTF-8 encoded filename* field
-        assertContentDisposition("\uD83E\uDD8A + x.jpg",
-                "attachment; filename=\"_.jpg\"; filename*=utf-8'en'%F0%9F%A6%8A%20+%20x.jpg");
-        assertContentDisposition("filename 的副本.jpg",
-                "attachment;filename=\"_.jpg\";" +
-                "filename*=UTF-8''filename%20%E7%9A%84%E5%89%AF%E6%9C%AC.jpg");
-        assertContentDisposition("filename.jpg",
-                "attachment; filename=_.jpg; filename*=utf-8'en'filename.jpg");
-
-        // ISO-8859-1 encoded filename* field
-        assertContentDisposition("file' 'name.jpg",
-                "attachment; filename=\"_.jpg\"; filename*=iso-8859-1'en'file%27%20%27name.jpg");
+
+        for (int i = 0; i < CONTENT_DISPOSITION_TYPES; i++) {
+
+            if (i == CONTENT_DISPOSITION_TYPES - 1) {
+                contentDisposition = "inline";
+            }
+
+            //continuing with default filenames
+            assertContentDisposition("downloadfile.bin", contentDisposition);
+            assertContentDisposition("downloadfile.bin", contentDisposition + ";");
+            assertContentDisposition("downloadfile.bin", contentDisposition + "; filename");
+            assertContentDisposition(".bin", contentDisposition + "; filename=");
+            assertContentDisposition(".bin", contentDisposition + "; filename=\"\"");
+
```

Comment:
\[PMD\] INFO: The String literal "filename\.jpg" appears 7 times in this file; the first occurrence is on line 44